tor_core/heartbeat.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+import cherrypy
+
+from tor_core.config import config
+
+conf = {
+    '/': {
+        'request.dispatch': cherrypy.dispatch.MethodDispatcher(),
+        'tools.response_headers.on': True,
+        'tools.response_headers.headers': [('Content-Type', 'application/json')],
+    }
+}
+
+
+def configure_heartbeat(config):
+    """
+    Sets up and starts the cherrypy server that operates the json api endpoint
+    for the heartbeat. Usage:
+
+    >>> import requests
+    >>> requests.get('http://localhost:{portnumber}').json()
+    returns:
+    {
+        'bot_name': 'this_is_an_awesome_bot',
+        'bot_version': '9001',
+        'core_version': '0.2.0'
+    }
+
+    :param config: the global config object
+    :return: None
+    """
+
+    # update the global config (separate from the application config above)
+    cherrypy.config.update(
+        {
+            'server.socket_port': config.heartbeat_port
+        }
+    )
+    logging.info('Heartbeat port: {}'.format(config.heartbeat_port))
+
+    if config.heartbeat_logging is False:  # defaults to false
```

Comment:
I perceive that `if config.heartbeat_logging is False` is clearer; it's not clear what `config.heartbeat_logging` is otherwise, even if it's not really important.